File: examples/update.rs
//! SPARQL Update example for Oxiland 0.3.
//!
//! Run with `cargo run --example update`.

use oxiland::{Model, Query, QueryResults, Update};

fn main() -> oxiland::Result<()> {
    let model = Model::new()?;
    Update::new("INSERT DATA { <https://example.com/alice> <https://example.com/name> \"Alice\" }")
        .execute(&model)?;

    let results =
        Query::new("ASK { <https://example.com/alice> <https://example.com/name> \"Alice\" }")
            .execute(&model)?;
    assert!(matches!(results, QueryResults::Boolean(true)));

    Update::new("DELETE DATA { <https://example.com/alice> <https://example.com/name> \"Alice\" }")
        .execute(&model)?;
    assert!(model.is_empty()?);
    Ok(())
}

File: python/examples/update.py
"""SPARQL Update INSERT/DELETE over an in-memory model."""

from oxiland import Model, query, update

model = Model()
update(
    model,
    'INSERT DATA { <https://example.com/alice> <https://example.com/name> "Alice" }',
)
assert query(
    model,
    'ASK { <https://example.com/alice> <https://example.com/name> "Alice" }',
) is True
update(
    model,
    'DELETE DATA { <https://example.com/alice> <https://example.com/name> "Alice" }',
)
assert model.is_empty()
print("update ok")
